ee/agent/reset.go

```diff
@@ -0,0 +1,332 @@
+// depending on launcher configuration, and then parses the tenant munemo from it.
+func currentMunemo(k types.Knapsack) (string, error) {
+	var enrollSecret string
+	if k.EnrollSecret() != "" {
+		enrollSecret = k.EnrollSecret()
+	} else if k.EnrollSecretPath() != "" {
+		content, err := os.ReadFile(k.EnrollSecretPath())
+		if err != nil {
+			return "", fmt.Errorf("could not read secret at enroll_secret_path %s: %w", k.EnrollSecretPath(), err)
+
+		}
+		enrollSecret = string(bytes.TrimSpace(content))
+	} else {
+		return "", errors.New("enroll secret and secret path both unset")
+	}
+
+	// We cannot verify since we don't have the key, so we use ParseUnverified
+	token, _, err := new(jwt.Parser).ParseUnverified(enrollSecret, jwt.MapClaims{})
+	if err != nil {
+		return "", fmt.Errorf("parsing enroll secret jwt: %w", err)
+	}
+
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return "", errors.New("enroll secret has no claims")
+	}
+
+	munemo, ok := claims["organization"]
+	if !ok {
+		return "", errors.New("enroll secret claims missing organization claim")
+	}
+
+	munemoStr, ok := munemo.(string)
+	if !ok {
+		return "", errors.New("munemo is unsupported type")
+	}
+
+	return munemoStr, nil
+}
+
+// takeDatabaseBackup retrieves the data we want to preserve from various db stores
+// as a snapshot of this db, appends it to previous snapshots if they exist, and
+// returns the collection of backup data.
+func takeDatabaseBackup(k types.Knapsack) ([]byte, error) {
+	nodeKey, err := k.ConfigStore().Get([]byte("nodeKey"))
+	if err != nil {
+		k.Slogger().Log(context.TODO(), slog.LevelWarn, "could not get node key from store", "err", err)
+	}
+
+	localPubKey, err := getLocalPubKey(k)
+	if err != nil {
+		k.Slogger().Log(context.TODO(), slog.LevelWarn, "could not get local pubkey from store", "err", err)
+	}
+
+	serial, err := k.HostDataStore().Get(hostDataKeySerial)
+	if err != nil {
+		k.Slogger().Log(context.TODO(), slog.LevelWarn, "could not get serial from store", "err", err)
+	}
+
+	hardwareUuid, err := k.HostDataStore().Get(hostDataKeyHardwareUuid)
+	if err != nil {
+		k.Slogger().Log(context.TODO(), slog.LevelWarn, "could not get hardware uuid from store", "err", err)
+	}
+
+	munemo, err := k.HostDataStore().Get(hostDataKeyMunemo)
+	if err != nil {
+		k.Slogger().Log(context.TODO(), slog.LevelWarn, "could not get munemo from store", "err", err)
+	}
+
+	deviceId, err := k.ServerProvidedDataStore().Get([]byte("device_id"))
+	if err != nil {
+		k.Slogger().Log(context.TODO(), slog.LevelWarn, "could not get device id from store", "err", err)
+	}
+
+	remoteIp, err := k.ServerProvidedDataStore().Get([]byte("remote_ip"))
+	if err != nil {
+		k.Slogger().Log(context.TODO(), slog.LevelWarn, "could not get remote ip from store", "err", err)
+	}
+
+	tombstoneId, err := k.ServerProvidedDataStore().Get([]byte("tombstone_id"))
+	if err != nil {
+		k.Slogger().Log(context.TODO(), slog.LevelWarn, "could not get tombstone id from store", "err", err)
```

Comment:
do we think it would be worth it to wire the parent context through this method for these logs? (instead of all the new TODOs)

Reply:
Done in https://github.com/kolide/launcher/pull/1492/commits/d963880bf33469ec2126439d074a2b4e995ce703